xcdat/spatial.py

```diff
@@ -441,12 +408,34 @@ def _get_longitude_weights(
         -------
         xr.DataArray
             The longitude axis weights.
+
+        Raises
+        ------
+        ValueError
+            If the there are multiple instances in which the
+            domain_bounds[:, 0] > domain_bounds[:, 1]
         """
         p_meridian_index: Optional[np.ndarray] = None
         d_bounds = domain_bounds.copy()
 
+        # check if there is more than one potential prime meridian cell
+        # there should only be one for rectilinear data
+        pmcells = np.where(domain_bounds[:, 1] - domain_bounds[:, 0] < 0)[0]
+        if len(pmcells) > 1:
+            raise ValueError(
+                "More than one longitude bound is out of order. Only one bound \n\
+                              spanning the prime meridian is permitted"
+            )
+        # convert longitude bounds to 360 degree domain
+        d_bounds: xr.DataArray = self._swap_lon_axis(d_bounds, to=360)  # type: ignore
+        # check for bounds spanning prime meridian
+        p_meridian_index = _get_prime_meridian_index(d_bounds)
+        # if bounds span a prime meridian, ensure bounds are organized to span 0-360
+        # (without losing weight)
+        if p_meridian_index is not None:
+            d_bounds = _align_lon_bounds_to_360(d_bounds, p_meridian_index)
+
```

Comment:
I pulled these lines of code out of the `if region_bounds is not None` block (below). This logic should apply whether there is a `region_bound` provided or not, because it helps to ensure the longitude bounds span 0 to 360 and are linear (high-to-low) which is important during the weights calculation.

Reply:
This was originally put in this if statement, because the user could pass in region bounds in (-180, 180) or (0, 360). If the user passed in region bounds, we converted both to a (0, 360) domain and then handled prime meridian issues. 

The issue prompting this shows that for some datasets, we need to deal with the prime meridian issue (even if there are no region bounds). This doesn't occur in CMIP data (I think because CF-conventions state that bounds should be increasing). If bounds are increasing, there is no need for this complex logic.